uw-research-computing/gpu-jobs.md

```diff
@@ -363,23 +252,124 @@ option to your submit file:
+server, including:
+
+<table class="gtable">
+	<tr>
+		<th>Attribute </th>
+		<th>Explanation </th>
+	</tr>
+	<tr>
+		<td><code>Gpus</code></td>
+		<td>Number of GPUs in an individual job slot on a server (one server can be divided into slots to run multiple jobs).</td>
+	</tr>
+	<tr>
+		<td><code>TotalGPUs</code></td>
+		<td>The total number of GPUs on a server.</td>
+	</tr>
+	<tr>
+		<td><code>CUDADeviceName</code></td>
+		<td>The type of GPU card.</td>
+	</tr>
+	<tr>
+		<td><code>CUDACapability</code></td>
+		<td>Represents various capabilities of the GPU. Can be used as a proxy for the GPU card type when 
+		requiring a specific type of GPU. <a href="https://en.wikipedia.org/wiki/CUDA#GPUs_supported">Wikipedia</a>
+		has a table showing the CUDA compute capability for specific GPU architectures and cards.
+		More details on what the capability numbers mean can be found on the 
+		<a href="https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#compute-capabilities">
+		NVIDIA website</a>.</td>
+	</tr>
+	<tr>
+		<td><code>CUDADriverVersion</code></td>
+		<td>Maximum CUDA runtime version supported by the GPU drivers on the server. </td>
+	</tr>
+	<tr>
+		<td><code>CUDAGlobalMemoryMb</code></td>
+		<td>Amount of memory available on the GPU card.</td>
+	</tr>
+</table>
+
+# E. Preparing Software Using GPUs
+
+Before using GPUs in CHTC you should ensure that the use of GPUs will
+actually help your program run faster. This means that the code or
+software you are using has the special programming required to use GPUs
+and that your particular task will use this capability.
+
+If this is the case, there are several ways to run GPU-enabled software
+in CHTC:
+
+> **Machine Learning**\
+>  For those using machine learning code specifically, we have a guide
+> with more specific recommendations here: [Run Machine Learning Jobs on
+> HTC](machine-learning-htc.html)
+
+## 1. Compiled Code
+
+You can use our conventional methods of creating a portable installation
+of a software package (as in our R/Python guides) to run on GPUs. Most
+of our build servers or GPU servers have copies of the CUDA Runtime that
+can be used to compile code. To access these servers, submit an
+interactive job, following the instructions in our [Build Job
+Guide](inter-submit.html) or by submitting a GPU job submit file with the
+interactive flag for `condor_submit`. Once on a build or GPU server, see
+what CUDA versions are available by looking at the path
+`/user/local/cuda-*`.
+
+Note that we strongly recommend software installation strategies that
+incorporate the CUDA runtime into the final installed code, so that jobs
+are able to run on servers even if a different version of the CUDA
+runtime is installed (or there's no runtime at all!). For compiled code,
+look for flags that enable static linking or use one of the solutions
+listed below.
+
+## 2. Docker
+
+CHTC's GPU servers have "nvidia-docker" installed, a specific version of
```

Comment:
Has nvidia-docker been rebranded?  https://github.com/NVIDIA/nvidia-docker now refers to the "NVIDIA Container Toolkit", though the repo name hasn't changed.

Reply:
This is a good question. I don't really know the state of Docker and GPUs!

Reply:
This isn't really critical for this pull request.  I'll resolve the comment.